src/pudl/analysis/allocate_net_gen.py

```diff
@@ -622,31 +621,23 @@ def associate_generator_tables(
             on=IDX_PM_ESC,
             how="outer",
             validate="m:1",
-            indicator=True,  # used in _allocate_unassociated_records to find unassocited
         )
         .pipe(remove_inactive_generators)
-        .pipe(
-            _allocate_unassociated_records,
-            idx_cols=IDX_PM_ESC,
-            col_w_unexpected_codes="prime_mover_code",
-            data_columns=[f"{col}_gf_tbl" for col in DATA_COLUMNS],
-        )
-        .drop(columns=["_merge"])  # drop do we can do this again in the bf_summed merge
         .merge(
             bf_by_gens, on=IDX_GENS_PM_ESC, how="outer", validate="m:1", indicator=True
         )
         .pipe(
-            _allocate_unassociated_records,
+            _allocate_unassociated_bf_records,
             idx_cols=IDX_GENS_PM_ESC,
-            col_w_unexpected_codes="energy_source_code",
+            col_w_unexpected_codes="prime_mover_code",
```

Comment:
So here you are focused on matching unassociated prime movers, but it seems like the `_allocate_unassociated_bf_records` also has some code related to energy source codes? If this function is no longer being used to associate both prime movers and energy source codes, I'm wondering if it should be tailored to specifically associate prime mover codes in the bf table?

Reply:
I believe some of the energy_source_code_num stuff in the function was to aid in associating the non-matching ESC record's data to *only* the primary energy source code record instead of sloshing it across the full plant.

But you are very right that this should be removed. I tested it a few times and tailoring this to the PM-codes only seems to work great.